modules/glue/iam.tf

```diff
@@ -1,4 +1,8 @@
 ####Glue retraining Job IAM role##########
+locals {
```

Comment:
Constructing the list of buckets and keys as locals supports both of the following cases:
1. data_bucket is not encrypted, so KMS key is not added to list
2. data_bucket is encrypted, so KMS key is added to list, and permissions added to IAM policy